src/idl/src/tree.c

```diff
@@ -918,6 +937,21 @@ bool idl_is_struct(const void *ptr)
   return true;
 }
 
+bool idl_is_empty_struct(const void *node)
+{
+  if (!idl_is_struct(node))
+    return false;
+  const idl_struct_t *s = (const idl_struct_t *)node;
+  if (!s->members)
+    return true;
+  bool empty = true;
+  for (idl_member_t *m = s->members; empty && m; m = idl_next(m)) {
+    if (!idl_is_empty_struct(m->type_spec))
+      empty = false;
+  }
```

Comment:
A struct that is empty will be omitted from the generated type support output (because empty structs are not allowed in c99), and therefore a member of an empty type should also be omitted. This also applies when using nested types, if the inner type does not have any member, all parents in the tree (that have no other members and no base type) can't be included in the output.